docs/integrations/app-development/opentelemetry/puppet-opentelemetry.md

```diff
@@ -0,0 +1,234 @@
+if [ ! -e last_run_date.tmp ]
+then
+date "+%Y-%m-%d" --date=1970-01-01 > last_run_date.tmp;
+fi
+find . -maxdepth 2 -type f -name "*.yaml" -newermt "$(cat last_run_date.tmp)" |
+while read f;
+do
+if [ -s "$f" ]
+then
+sed 1d "$f" > tmpfile;
+ruby -rjson -ryaml -e "resource_list = ''
+json = JSON.generate({});data = JSON.parse(json);
+obj = JSON.parse(YAML.load_file('tmpfile').to_json);
+data['host'] = obj['host'];
+data['time'] = obj['time'];
+data['configuration_version'] = obj['configuration_version'];
+data['transaction_uuid'] = obj['transaction_uuid'];
+data['report_format'] = obj['report_format'];
+data['puppet_version'] = obj['puppet_version'];
+data['status'] = obj['status'];
+data['transaction_completed'] = obj['transaction_completed'];
+data['noop'] = obj['noop'];
+data['noop_pending'] = obj['noop_pending'];
+data['environment'] = obj['environment'];
+data['metrics'] = obj['metrics'];
+data['resource_list'] = resource_list;
+i = 0;
+obj['resource_statuses'].each{
+ |j|
+if i == 0 then
+ tempdata = j[0];
+else
+ tempdata = ',' + j[0];
+end
+data['resource_list'] << tempdata;
+i = i + 1;
+};
+puts data.to_json;
+obj['resource_statuses'].each{
+ |y|  logsres = y[1];
+ logsres['transaction_uuid'] = obj['transaction_uuid'];
+ logsres['environment'] = obj['environment'];
+ logsres['host'] = obj['host'];
+ puts logsres.to_json;
+ };
+obj['logs'].each {
+ |x|logsjson = x;
+ logsjson['host'] = obj['host'];
+ logsjson['transaction_uuid'] = obj['transaction_uuid'];
+ logsjson['environment'] = obj['environment'];
+ puts logsjson.to_json
+ };"
+echo "$f conversion status:" $? >> "$log_file_name"
+date "+%d-%b-%Y %H:%M:%S" > last_run_date.tmp
+fi
+done
+echo "Conversion Completed. End Time: $(date)" >> "$log_file_name"
+```
+
+Please copy this script to `/opt/puppetlabs/server/data/puppetserver/reports` with the name puppetReport.sh. If Puppet is installed in a different location, copy the script to the correct path.
+
+Configure a cron job to trigger the python script using crontab. Frequency of this job can be set following the instructions from [here](https://www.python-engineer.com/posts/cron-jobs-for-python/#crontab). Here is the command which needs to be configured as part of cron to trigger the script:
+```sh
+<frequency_expression> /opt/puppetlabs/server/data/puppetserver/reports/puppetReport.sh
+```
+Please modify the location of the `puppetReport.sh` if required in the above command. The execution of the script above will generate a log file named `puppet_rpt_conversion.log`. The path to this log file needs to be provided during the app installation.
+
+## Collection configuration & App installation
+
+As part of the setting up the collection process and app installation user can select the App from App Catalog and click on **Install App**. Please follow the steps below.
+
+### Step 1: Set up Collector
+
+If you want to use an existing Otel Collector then this step can be skipped by selecting the option of using an existing Collector.
```

Comment:
Please follow the same convention as in other docs. You can refer any db or webserver based logs.